### ISSUE TITLE
Not working with custom Skelleton

### ISSUE THREAD

**Issue by lotsofbugs** [hacoo/ikmod#8]
Compiled it for 4.21 with success. The demo works just fine.
I didn't want to migrate all the unnecessary stuff to my project, so I copied the plugin, all the nodes inside the AnimBP, and the variables inside AnimBP and CharacterBP. 
I set up all the variables properly, but it isn't working. 

Any idea of what it might be?

**Comment by lotsofbugs** [comment]
I even tried to migrate the animBP CharPatroller with all its dependencies to my game, but it doesn't works either

**Comment by hacoo** [comment]
I'm trying to see if I can replicate your issue. Do you see any errors or warnings in the output log?

**Comment by lotsofbugs** [comment]
LogRTIK: Warning: Could not initialize FAnimNode_RangeLimitedFabrik -- An input wrapper object was null
LogRTIK: Warning: Could not initialize FAnimNode_RangeLimitedFabrik -- An input wrapper object was null
LogRTIK: Warning: Coud not initialize humanoid arm torso adjust - An input wrapper object was null
LogRTIK: Warning: Could not initialize Humanoid Foot Rotation Controller-- An input wrapper object was null
LogRTIK: Warning: Could not initialize Humanoid Foot Rotation Controller-- An input wrapper object was null
LogRTIK: Warning: Could not initialize Humanoid IK Leg Knee Correction -- An input wrapper object was null
LogRTIK: Warning: Could not initialize Humanoid IK Leg Knee Correction -- An input wrapper object was null
LogRTIK: Warning: Could not initialize Humanoid IK Leg -- An input wrapper object was null
LogRTIK: Warning: Could not initialize Humanoid IK Leg -- An input wrapper object was null
LogRTIK: Warning: Could not initialize biped hip adjustment -- one of the bone wrappers was null
LogRTIK: Warning: Could not initialize Humanoid IK Leg Trace -- Leg invalid
LogRTIK: Warning: Could not initialize Humanoid IK Leg Trace -- Leg invalid

**Comment by lotsofbugs** [comment]
But as I could see in you demo, these erros also happen there

**Comment by hacoo** [comment]
Yes, I see those warnings too (I should probably fix that). That log output looks OK.

I just tool a look at CHAR_Patroller, that blueprint is NOT configured to use IK at all! Try one of the characters from the demo scene --  CHAR_RiflePatroller or CHAR_PatrolMannequin. Place one of them on a non-flat surface, and see if the feet IK onto it.

**Comment by lotsofbugs** [comment]
You were right. I migrated CHAR_PatrolMannequin right now, and it works. But the problem with my custom skelleton persists

**Comment by hacoo** [comment]
Ok! This is probably a setup issue.

First, create a scene with your custom character ONLY. What log output do you see?

**Comment by lotsofbugs** [comment]
Only this: 

LogRTIK: Warning: Could not initialize FAnimNode_RangeLimitedFabrik -- An input wrapper object was null
LogRTIK: Warning: Could not initialize FAnimNode_RangeLimitedFabrik -- An input wrapper object was null
LogRTIK: Warning: Coud not initialize humanoid arm torso adjust - An input wrapper object was null
LogRTIK: Warning: Could not initialize Humanoid Foot Rotation Controller-- An input wrapper object was null
LogRTIK: Warning: Could not initialize Humanoid Foot Rotation Controller-- An input wrapper object was null
LogRTIK: Warning: Could not initialize Humanoid IK Leg Knee Correction -- An input wrapper object was null
LogRTIK: Warning: Could not initialize Humanoid IK Leg Knee Correction -- An input wrapper object was null
LogRTIK: Warning: Could not initialize Humanoid IK Leg -- An input wrapper object was null
LogRTIK: Warning: Could not initialize Humanoid IK Leg -- An input wrapper object was null
LogRTIK: Warning: Could not initialize biped hip adjustment -- one of the bone wrappers was null
LogRTIK: Warning: Could not initialize Humanoid IK Leg Trace -- Leg invalid
LogRTIK: Warning: Could not initialize Humanoid IK Leg Trace -- Leg invalid

**Comment by hacoo** [comment]
Did you intialize your IK chains? 

Take a look at the event graph (not animgraph!) portion of the provided AnimBPs. You should see a bunch of initialization logic, this stuff needs to be copied too! Check the screenshot below.

![image](https://user-images.githubusercontent.com/12739249/56097263-ca1d2300-5ea6-11e9-94d9-5dd134c3cceb.png)

**Comment by lotsofbugs** [comment]
Yes, I copied them into my AnimBP. Also copied the AnimGraph.

![image](https://user-images.githubusercontent.com/23303304/56097284-8b986000-5ec8-11e9-9bad-91d8d8cf2843.png)

**Comment by hacoo** [comment]
OK. I'm digging into this to see what the problem could be.

Just to confirm, did you set all of the bone references in the 'IK chains' and 'IK bones' variables to bones in your skeleton?

Can you send me a screenshot of the Animgraph part of your BP as well?

**Comment by lotsofbugs** [comment]
Yes, I've set the references of the bones and chain variables:

![image](https://user-images.githubusercontent.com/23303304/56097514-a9b38f80-5ecb-11e9-89ea-93938265b391.png)

**Comment by hacoo** [comment]
Sorry this isn't working for you. I'm not sure how to debug this without looking at your animBP more closely.

One thing you could try would be to make a copy of CHAR_PatrolMannequin, and set all the variables to use your bones. My best guess is that something wasn't copied correctly from the original animBP, that might help track down what is different.

**Comment by lotsofbugs** [comment]
Copied CHAR_PatrolMannequin and only changed the bones variables. Didn't work. Maybe is a problem with my skelleton and the bone's references I'm using. Do you mind taking a look into it? I can send him and a mesh to test.

**Comment by hacoo** [comment]
Yes, I can take a look. It seems like you're doing everything correctly, so I'm not sure what's wrong.

If you email me your skeleton and mesh, I'll see if I can get it to work. My email is hacoo36@gmail.com

**Comment by fire** [comment]
In 4.22, the code does get actor rather than get pawn and the cast fails. Result is that for me the arm ik doesn't work.

**Comment by lotsofbugs** [comment]
Revisited this issue today and I still can't manage it to work.

**Comment by fire** [comment]
I was able to compile this and get it working on the mac.

**Comment by fire** [comment]
https://github.com/fire/rtik You'll have to wait for the upload, but these are my changes.

**Comment by lotsofbugs** [comment]
> I was able to compile this and get it working on the mac.

I was also able to compile and run it, the problem is setting up a custom skeleton.

With the average Max Pelvis Adjust Height of 60 I could only achieve the feet right rotation on slopes.
I increased the Max Pelvis Adjust Height to 120 and managed to achieve some IK simulation, but now the mesh gets a weird location at the vertical axis:

![image](https://user-images.githubusercontent.com/23303304/77567533-0c682600-6ea6-11ea-87dd-4857b3e58e11.png)
